databuilder/extractor/presto_view_metadata_extractor.py

```diff
@@ -0,0 +1,132 @@
+
+        self.sql_stmt = PrestoViewMetadataExtractor.SQL_STATEMENT.format(
+            where_clause_suffix=conf.get_string(PrestoViewMetadataExtractor.WHERE_CLAUSE_SUFFIX_KEY))
+
+        LOGGER.info('SQL for hive metastore: {}'.format(self.sql_stmt))
+
+        self._alchemy_extractor = SQLAlchemyExtractor()
+        sql_alch_conf = Scoped.get_scoped_conf(conf, self._alchemy_extractor.get_scope())\
+            .with_fallback(ConfigFactory.from_dict({SQLAlchemyExtractor.EXTRACT_SQL: self.sql_stmt}))
+
+        self._alchemy_extractor.init(sql_alch_conf)
+        self._extract_iter = None  # type: Union[None, Iterator]
+
+    def extract(self):
+        # type: () -> Union[TableMetadata, None]
+        if not self._extract_iter:
+            self._extract_iter = self._get_extract_iter()
+        try:
+            return next(self._extract_iter)
+        except StopIteration:
+            return None
+
+    def get_scope(self):
+        # type: () -> str
+        return 'extractor.presto_view_metadata'
+
+    def _get_extract_iter(self):
+        # type: () -> Iterator[TableMetadata]
+        """
+        Using itertools.groupby and raw level iterator, it groups to table and yields TableMetadata
+        :return:
+        """
+        row = self._alchemy_extractor.extract()
+        while row:
+            columns = self._get_column_metadata(row['view_original_text'])
+            yield TableMetadata(database='presto',
+                                cluster=self._cluster,
+                                schema_name=row['schema_name'],
+                                name=row['name'],
+                                description=None,
+                                columns=columns,
+                                is_view=True)
+            row = self._alchemy_extractor.extract()
+
+    def _get_column_metadata(self, view_original_text):
+        # type: (str) -> List[ColumnMetadata]
+        """
+        Get Column Metadata from VIEW_ORIGINAL_TEXT from TBLS table for Presto Views
+        Columns are sorted alphabetically
+        :param view_original_text:
+        :return:
+        """
+        encoded_view_info = (
+            view_original_text.
+            split(PrestoViewMetadataExtractor.PRESTO_VIEW_PREFIX, 1)[-1].
+            rsplit(PrestoViewMetadataExtractor.PRESTO_VIEW_SUFFIX, 1)[0]
+        )
+        decoded_view_info = self._decode_base64(encoded_view_info)
+        sorted_view_column_info = self._sort_columns_alphabetically(json.loads(decoded_view_info).get('columns'))
+        columns = []
+        for i in range(len(sorted_view_column_info)):
+            column = sorted_view_column_info[i]
+            columns.append(ColumnMetadata(name=column['name'],
+                                          description=None,
+                                          col_type=column['type'],
+                                          sort_order=i))
+        return columns
+
+    def _sort_columns_alphabetically(self, columns):
```

Comment:
I think we should allow user to specific the sorting algorithm, and default to sort with alphabetically

E.g we could specify a config key to define lambda sort function, with default value as lambda k: k['name'].

Reply:
@feng-tao How you want this to be configured? The key in `sorted(columns, key=func)` is a function. You want user to pass a function through pyhocon ConfigFactory as a String and then we interpret it to a function? This does not sound like a good example.

Reply:
WDYT @jinhyukchang , could pyhocon take a callable ? I am thinking whether we could make the sorting function as a config from user with default using alphabetically order. But if that is not possible, I am ok with the current approach.

Reply:
+1 Not sure sorting alphabetically make sense in general. Is the original payload JSON array? If that is the case, does it show same order as it appears in SELECT statement?

Reply:
So based on our experiment, it is JSON array and show same order as it appears in SELECT statement. Will remove alphabetically sorting and keep it original way.